src/User/UsernameSlugDriver.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\User;
+
+use Flarum\Database\AbstractModel;
+use Flarum\Http\SlugDriverInterface;
+
+class UsernameSlugDriver implements SlugDriverInterface
+{
+    public function toSlug(AbstractModel $instance): string
+    {
+        return $instance->username;
+    }
+
+    public function fromSlug(string $slug, User $actor): AbstractModel
+    {
+        return User::where('username', $slug)->whereVisibleTo($actor)->firstOrFail();
```

Comment:
Again, I think this would be better as a method on UserRepository

Reply:
I agree, I'll update this